### DISCUSSION TITLE
Set SOURCE in ERROR without THROW()

### DISCUSSION THREAD

**Discussion by mpapini64** [nextgenhealthcare/connect#5413] [Q&A] [answered]
New question ... 
I have a channel whose SOURCE is a FILE READER, I must be able to put the SOURCE in an error state so that it moves the file under analysis in the folder that I have dedicated to errors.
I would like to avoid calling the `throw ("xxx")`, because that way I don't see the error in the metadata, but simply that that channel has failed on that line.
Summarizing in RESPONSE I can use `responseStatus= ERROR`, but in a SOURCE or DESTINATION how can I set the error status without necessarily forcing an exception through THROW ()

**Comment by mpapini64** [comment, marked as answer]
I know that cmq does not work, regardless of whether it goes in error or not that file is moved to the folder of the documents so nothing, sorry I do not need to set the SOURCE in error!

**Reply by pacmano1** [reply]
Not clear on the flow here, but I would not use the file move stuff on the source tab and instead use java to move the file around as needed.

**Reply by mpapini64** [reply]
Yes, I realized it in the end!
I decided to have the original file deleted after processing as the "File Reader" setting, while the two destinations copy respectively to the .. \ OK or .. \ BAD folder depending on whether the file is correct or not.
